src/test.ts

```diff
@@ -46,17 +48,42 @@ export function runTest(testDirectory: string, rulesDirectory?: string | string[
 
     for (const fileToLint of filesToLint) {
         const fileBasename = path.basename(fileToLint, FILE_EXTENSION);
+        const fileCompileName = fileBasename.replace(/\.lint$/, "");
         const fileText = fs.readFileSync(fileToLint, "utf8");
         const fileTextWithoutMarkup = parse.removeErrorMarkup(fileText);
         const errorsFromMarkup = parse.parseErrorsFromMarkup(fileText);
 
+        const compilerOptions = createCompilerOptions();
+        const compilerHost: ts.CompilerHost = {
+            fileExists: () => true,
+            getCanonicalFileName: (filename: string) => filename,
+            getCurrentDirectory: () => "",
+            getDefaultLibFileName: () => ts.getDefaultLibFileName(compilerOptions),
+            getNewLine: () => "\n",
+            getSourceFile: function (filenameToGet: string) {
+                if (filenameToGet === this.getDefaultLibFileName()) {
+                    const fileText = fs.readFileSync(ts.getDefaultLibFilePath(compilerOptions)).toString();
+                    return ts.createSourceFile(filenameToGet, fileText, compilerOptions.target);
+                } else if (filenameToGet === fileCompileName) {
+                    return ts.createSourceFile(fileBasename, fileTextWithoutMarkup, compilerOptions.target, true);
+                }
+            },
+            readFile: () => null,
+            useCaseSensitiveFileNames: () => true,
+            writeFile: () => null,
+        };
+
+        const program = ts.createProgram([fileCompileName], compilerOptions, compilerHost);
+        // perform type checking on the program, updating nodes with symbol table references
```

Comment:
Not sure, but in this case, the tests just have to be parsed and type checked as much as possible, ignoring compiler errors.